server/svix-server/src/worker.rs

```diff
@@ -193,20 +193,24 @@ async fn dispatch(
                 response_status_code: Set(status_code),
                 response: Set(body),
                 status: Set(status),
-
                 ..attempt
             };
-            Ok(attempt)
+            match status {
+                MessageStatus::Success => Ok(attempt),
+                _ => Err((
+                    attempt,
+                    crate::error::Error::Generic(format!("HTTP STATUS {}", status_code)),
+                )),
+            }
         }
         Err(err) => {
             let attempt = messageattempt::ActiveModel {
                 response_status_code: Set(0),
                 response: Set("".to_owned()),
                 status: Set(MessageStatus::Fail),
-
                 ..attempt
             };
-            Err((attempt, err))
+            Err((attempt, crate::error::Error::Generic(format!("{}", err))))
```

Comment:
What was wrong with what was there before?

Reply:
This is fixed -- was using as a placeholder b/c I was struggling w/ getting a reqwest::Error in the http-non-200 scenario.